eldritch/assets.py

```diff
@@ -273,6 +273,44 @@ def get_trigger(self, event, owner, state):
     return None
 
 
+class BlessingOrCurse(Card):
+  def __init__(self, kind, idx):
+    assert kind in ["Blessing", "Curse"]
+    super().__init__(kind, idx, "specials", {}, {})
+    self.opposite = "Curse" if kind == "Blessing" else "Blessing"
+    self.must_roll = False
+
+  def get_trigger(self, event, owner, state):
+    if isinstance(event, events.UpkeepActions) and event.character == owner:
+      if self.must_roll:
```

Comment:
The other thing to look out for here is that must_roll does not reset if you discard a blessing/curse. So you might have to roll if you draw a blessing that someone else had already used and discarded.